hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaArtifactManagement.java

```diff
@@ -141,6 +144,21 @@ private void assertMaxArtifactSizeQuota(final String filename, final long id, fi
         }
     }
 
+    private void assertMaxArtifactSizeTotalQuota(final String filename, final long artifactSize) {
+        final long maxArtifactSizeTotal = quotaManagement.getMaxArtifactStorage();
+        if (maxArtifactSizeTotal <= 0) {
+            return;
+        }
+
+        final Optional<Long> currentlyUsed = localArtifactRepository.getSumOfUndeletedArtifactSize();
+        if (currentlyUsed.isPresent() && currentlyUsed.get() + artifactSize > maxArtifactSizeTotal) {
```

Comment:
fix this ... if currentlyUsed is empty, we need to continue with '0' .... the quota could still be violated in this case...

Reply:
and add a test case ... ;-)

Reply:
Done